src/segment_media.go

```diff
@@ -0,0 +1,134 @@
+	MediaTrackSeparator Property = "track_separator"
+	// MediaTimeSeparator is put between the media position and total time
+	MediaTimeSeparator Property = "time_separator"
+	// MediaShowTime is media time show or hidden switch
+	MediaIsShowTime Property = "is_show_time"
+)
+
+type CShapTimeSpan struct {
+	Ticks             int     `json:"Ticks"`
+	Days              int     `json:"Days"`
+	Hours             int     `json:"Hours"`
+	Milliseconds      int     `json:"Milliseconds"`
+	Minutes           int     `json:"Minutes"`
+	Seconds           int     `json:"Seconds"`
+	TotalDays         float64 `json:"TotalDays"`
+	TotalHours        float64 `json:"TotalHours"`
+	TotalMilliseconds float64 `json:"TotalMilliseconds"`
+	TotalMinutes      float64 `json:"TotalMinutes"`
+	TotalSeconds      float64 `json:"TotalSeconds"`
+}
+
+type NowPlayingSessionInfo struct {
+	Session struct {
+		PID            int    `json:"PID"`
+		RenderDeviceID string `json:"RenderDeviceId"`
+		SourceAppID    string `json:"SourceAppId"`
+		SourceDeviceID string `json:"SourceDeviceId"`
+	} `json:"Session"`
+	Playback struct {
+		PropsValid          int       `json:"PropsValid"`
+		PlaybackCaps        int       `json:"PlaybackCaps"`
+		PlaybackState       int       `json:"PlaybackState"`
+		PlaybackMode        int       `json:"PlaybackMode"`
+		RepeatMode          int       `json:"RepeatMode"`
+		PlaybackRate        int       `json:"PlaybackRate"`
+		ShuffleEnabled      bool      `json:"ShuffleEnabled"`
+		LastPlayingFileTime time.Time `json:"LastPlayingFileTime"`
+	} `json:"Playback"`
+	MediaInfo struct {
+		AlbumArtist         string   `json:"AlbumArtist"`
+		AlbumTitle          string   `json:"AlbumTitle"`
+		Subtitle            string   `json:"Subtitle"`
+		Title               string   `json:"Title"`
+		Artist              string   `json:"Artist"`
+		MediaClassPrimaryID string   `json:"MediaClassPrimaryID"`
+		Genres              []string `json:"Genres"`
+		AlbumTrackCount     int      `json:"AlbumTrackCount"`
+		TrackNumber         int      `json:"TrackNumber"`
+	} `json:"MediaInfo"`
+	Timeline struct {
+		StartTime           CShapTimeSpan `json:"StartTime"`
+		EndTime             CShapTimeSpan `json:"EndTime"`
+		MinSeekTime         CShapTimeSpan `json:"MinSeekTime"`
+		MaxSeekTime         CShapTimeSpan `json:"MaxSeekTime"`
+		Position            CShapTimeSpan `json:"Position"`
+		PositionSetFileTime time.Time     `json:"PositionSetFileTime"`
+	} `json:"Timeline"`
+}
+
+type media struct {
+	props *properties
+	env   environmentInfo
+	info  NowPlayingSessionInfo
+	other string
+}
+
+func (s *media) enabled() bool {
+	tool := "sys-media-info"
+	if s.env.isWsl() {
+		tool += ".exe"
+	}
+	if s.env.hasCommand(tool) {
+		str, err := s.env.runCommand(tool, "--json")
+		if err == nil && str != "{}" {
+			json.Unmarshal([]byte(str), &s.info)
+			return true
+		}
+	}
+	players := [...]string{"cloudmusic.exe", "qqmusic.exe", "spotify.exe"}
```

Comment:
I wonder what the difference is between this and the Spotify implementation on Windows? As it seems to fetch more, which is really interesting tbh.